File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        set1 = {}
        set2 = {}
        result = []
        for n in nums1:
            if n not in set1:
                set1[n] = 1
            else:
                set1[n] += 1
        for n in nums2:
            if n not in set2:
                set2[n] = 1
            else:
                set2[n] += 1
        print set1
        print set2
        for n in set1:
            if n in set2:
                for unit in range(min(set1[n],set2[n])):
                    result.append(n)
        return result

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int index1 = 0;
        int index2 = 0;
        List<Integer> values = new ArrayList<>();
        while(index1 < nums1.length && index2 < nums2.length){
            if(nums1[index1] == nums2[index2]){
                values.add(nums1[index1]);
                index1++;
                index2++;
            }
            else if(nums1[index1] < nums2[index2]){
                index1++;
            }
            else if(nums1[index1] > nums2[index2]){
                index2++;
            }
        }
        
        int[] answer = new int[values.size()];
        for(int i=0; i < values.size(); i++){
            answer[i] = values.get(i);
        }
        return answer;
    }
}